nilearn/input_data/nifti_masker.py

```diff
@@ -31,8 +31,8 @@ def filter_and_mask(imgs, mask_img_, parameters,
                     memory_level=0, memory=Memory(cachedir=None),
                     verbose=0,
                     confounds=None,
-                    copy=True):
-
+                    copy=True,
+                    return_affine=True):
```

Comment:
The thing is that `filter_and_mask` returns the affine because we need it to do sanity checks (reminder, for performance reasons, the `check_niimg` is called in `filter_and_mask`). So, if we directly shelve the result of `filter_and_mask`, the affine will be stored together, which is not useful. Plus, if we need the affine, we will have to unshelve the object.

One solution may be to add a `shelving` parameter to `filter_and_mask` and directly use the shelving inside.